packages/e2e-tests/src/tests/PKPEthers.spec.ts

```diff
@@ -0,0 +1,861 @@
+    });
+  });
+
+  describe('SignedTypedDatav3', () => {
+    it('LitAction Session', async () => {
+      await signTypedDatav3(devEnv, getLitActionSessionSigs);
+    });
+
+    it('LitAction IPFS Session', async () => {
+      await signTypedDatav3(devEnv, getLitActionSessionSigsUsingIpfsId);
+    });
+
+    it('EOA Wallet', async () => {
+      await signTypedDatav3(devEnv, getEoaSessionSigs);
+    });
+
+    it('PKP Session', async () => {
+      await signTypedDatav3(devEnv, getEoaSessionSigs);
+    });
+  });
+
+  describe('Signed Typed Data v4', () => {
+    it('LitAction Session', async () => {
+      await signTypedDatav4(devEnv, getLitActionSessionSigs);
+    });
+
+    it('LitAction IPFS Session', async () => {
+      await signTypedDatav4(devEnv, getLitActionSessionSigsUsingIpfsId);
+    });
+
+    it('EOA Wallet', async () => {
+      await signTypedDatav4(devEnv, getEoaSessionSigs);
+    });
+
+    it('PKP Session', async () => {
+      await signTypedDatav4(devEnv, getEoaSessionSigs);
+    });
+  });
+
+  describe('Sign With AuthContext', () => {
+    it('LitAction Session', async () => {
+      await signWithAuthContext(devEnv, getLitActionSessionSigs);
+    });
+
+    it('LitAction IPFS Session', async () => {
+      await signWithAuthContext(devEnv, getLitActionSessionSigsUsingIpfsId);
+    });
+
+    it('EOA Wallet', async () => {
+      await signWithAuthContext(devEnv, getEoaSessionSigs);
+    });
+
+    it('PKP Session', async () => {
+      await signWithAuthContext(devEnv, getEoaSessionSigs);
+    });
+  });
+});
+
+const signMessage = async (
+  devEnv: TinnyEnvironment,
+  generator: (
+    devEnv: TinnyEnvironment,
+    person: TinnyPerson,
+    resources?: LitResourceAbilityRequest[]
+  ) => Promise<SessionSigsMap | undefined>
+): Promise<void> => {
+  const alice = await devEnv.createRandomPerson();
+  const eoaSessionSigs = await generator(devEnv, alice);
+
+  const pkpEthersWallet = new PKPEthersWallet({
+    litNodeClient: devEnv.litNodeClient!,
+    pkpPubKey: alice.pkp?.publicKey!,
+    controllerSessionSigs: eoaSessionSigs,
+  });
+
+  await pkpEthersWallet.init();
+
+  // -- test signMessage
+  try {
+    const res = await pkpEthersWallet.signMessage(alice.loveLetter);
+    expect(res).toBeDefined();
+  } catch (e) {
+    throw new Error('❌ Error: ' + (e as Error).message);
+  } finally {
+    devEnv.releasePrivateKeyFromUser(alice);
+  }
+};
+
+const ethTransaction = async (
+  devEnv: TinnyEnvironment,
+  generator: (
+    devEnv: TinnyEnvironment,
+    person: TinnyPerson,
+    resources?: LitResourceAbilityRequest[]
+  ) => Promise<SessionSigsMap | undefined>
+): Promise<void> => {
+  const alice = await devEnv.createRandomPerson();
+  const eoaSessionSigs = await generator(devEnv, alice);
+
+  const pkpEthersWallet = new PKPEthersWallet({
+    litNodeClient: devEnv.litNodeClient!,
+    pkpPubKey: alice.pkp?.publicKey!,
+    controllerSessionSigs: eoaSessionSigs,
+  });
+
+  await pkpEthersWallet.init();
```

Comment:
Wouldn't we want those ops to fail in an obvious way? It seems dangerous to catch those errors as it would then cause the test to enter a separate code path for error handling.